pkg/v1/remote/write.go

```diff
@@ -849,27 +849,48 @@ func WriteLayer(repo name.Repository, layer v1.Layer, options ...Option) (rerr e
 // should ensure that all blobs or manifests that are referenced by t exist
 // in the target registry.
 func Tag(tag name.Tag, t Taggable, options ...Option) error {
-	o, err := makeOptions(tag.Context(), options...)
+	return manifestReference(tag, t, options...)
+}
+
+// Digest adds a manifest from the given Taggable via PUT /v1/.../manifest/<digest>
+//
+// Notable implementations of Taggable are v1.Image, v1.ImageIndex, and
+// remote.Descriptor.
+//
+// If t implements MediaType, we will use that for the Content-Type, otherwise
+// we will default to types.DockerManifestSchema2.
+//
+// Digest does not attempt to write anything other than the manifest, so callers
+// should ensure that all blobs or manifests that are referenced by t exist
+// in the target registry.
+func Digest(digest name.Digest, t Taggable, options ...Option) error {
```

Comment:
I'd lean towards just `remote.Put` for this, I think. The existing `remote.Tag` is using "tag" as a verb here, not as a noun, so calling this `Digest` doesn't make a ton of sense.

`remote.Put` would also mirror `remote.Get`, `remote.Head`, and `remote.Delete`.

It should take a `name.Reference` as well, IMO (basically just exposing `manifestReference` as `Put`).

Reply:
Done. Take a look.